### PR TITLE
Implement the ability to set notification position

### LINKED ISSUE #613: Add a way to change the position of notifications
**Operating System**: fedora 24 64 bit
**Budgie version**: Budgie 10.2.7

**The issue encountered**
There is no way to change the position of notifications

**Steps to reproduce the issue**
mmm... Change the default position Budgie panel in the "panel" tab on Raven.

**Current result**
Although the panel is bottom, notifications continue to appear in the upper right corner

**Expected result**
Notifications are displayed alongside to the main panel (lower right corner when the panel is bottom) or add options in Raven to change position of notifications

### PR DESCRIPTION
**Pull request by EbonJaeger**
Users can choose which corner of the screen notifications will appear in.

Currently the setting for this is in the Style settings for Budgie; this can be moved if desired.
This was my first time really playing with settings, so any suggestions for improvement would be appreciated!

The default choice is the same as what Budgie currently does (notifications appear in the top right corner), and there is no change in the top positioning.

All four corners have been tested both by the `notify-send` command and by natural use (mainly, Discord).
image

Resolves #613

### CODE REVIEW

**Comment by yursan9** [comment]
Did you implement the fix for #1619, @EbonJaeger?

**Comment by EbonJaeger** [comment]
@yursan9 I did not. I'm not sure if that would be better done as a separate PR after this one is merged, assuming this gets accepted.

**Comment by danrobinson78** [comment]
I think that #1619 should be a separate PR as well, it's kind of a separate issue. Even if we allow notifications to position differently, the issue of accounting for whether there's a panel in that corner or not needs to be accounted for.

I guess the other option even if we don't want to enable the option for this PR would just be to anchor the notification to the corner of the screen where the tray is anyway.

File: src/panel/settings/settings_style.vala [hunk 1]
```diff
@@ -159,6 +185,25 @@ public class StylePage : Budgie.SettingsPage {
...
+    /**
+     * Get a user-friendly name for each position.
+     */
+    public string notification_position_to_display(Budgie.NotificationPosition position)
+    {
+        switch (position) {
+            case NotificationPosition.TOP_LEFT:
+                return _("Top Left");
+            case NotificationPosition.TOP_RIGHT:
+                return _("Top Right");
```
[the remark this answers is not in the document]
Comment (EbonJaeger) [reply src/panel/settings/settings_style.vala]: Good point.

Resolved with bae1fd3

File: src/raven/notifications_view.vala [hunk 2]
```diff
@@ -747,33 +754,96 @@ public class NotificationsView : Gtk.Box
...
+                } else { // This is the first nofication on the screen
+                    x = (rect.x + rect.width) - NOTIFICATION_SIZE;
+                    x -= BUFFER_ZONE; // Don't touch edge of the screen
+
+                    int height;
+                    window.get_size(null, out height); // Get the estimated height of the notification
+                    y = (rect.y + rect.height) - height - INITIAL_BUFFER_ZONE;
+                }
+                break;
+            default:
```
[the remark this answers is not in the document]
Comment (EbonJaeger) [reply src/raven/notifications_view.vala]: Also true.

Resolved with b4d330b

**Comment by EbonJaeger** [comment]
Sorry about the local config file sneaking in there. I momentarily forgot that it would get automatically added with `git commit`.

**Comment by JoshStrobl** [comment]
LGTM. But for realsies no more features in Budgie 10.5 okay? I don't want our translators murdering me for even more stuff they need to translate and fix :smile:

**Comment by EbonJaeger** [comment]
@JoshStrobl Haha to be honest, I didn't think you'd actually pull it for 10.5. Would you be okay with a little further refinement of the positions with regard to the user's panel position? That wouldn't require new translations.

**Pull request merged**